runtime/src/sorted_storages.rs

```diff
@@ -64,23 +64,20 @@ impl<'a> SortedStorages<'a> {
     // 1. each SnapshotStorage.!is_empty()
     // 2. SnapshotStorage.first().unwrap().get_slot() is unique from all other SnapshotStorage items.
     pub fn new(source: &'a [SnapshotStorage]) -> Self {
-        let slots = source
-            .iter()
-            .map(|storages| {
-                let first = storages.first();
-                assert!(first.is_some(), "SnapshotStorage.is_empty()");
-                let storage = first.unwrap();
-                storage.slot() // this must be unique. Will be enforced in new_with_slots
-            })
-            .collect::<Vec<_>>();
-        Self::new_with_slots(source.iter().zip(slots.iter()), None, None)
+        let slots = source.iter().map(|storages| {
+            let first = storages.first();
+            assert!(first.is_some(), "SnapshotStorage.is_empty()");
+            let storage = first.unwrap();
+            storage.slot() // this must be unique. Will be enforced in new_with_slots
+        });
+        Self::new_with_slots(source.iter().zip(slots.into_iter()), None, None)
     }
 
     /// create `SortedStorages` from 'source' iterator.
     /// 'source' contains a SnapshotStorage and its associated slot
     /// 'source' does not have to be sorted in any way, but is assumed to not have duplicate slot #s
-    pub fn new_with_slots<'b>(
-        source: impl Iterator<Item = (&'a SnapshotStorage, &'b Slot)> + Clone,
+    pub fn new_with_slots(
```

Comment:
Main change is getting rid of `&'b` on Slot. Simplifies the api and was unnecessary.